agent-control/src/sub_agent/on_host/command/command_os.rs

```diff
@@ -75,14 +76,47 @@ impl CommandOSNotStarted {
 ////////////////////////////////////////////////////////////////////////////////////
 
 impl CommandOSStarted {
-    pub(crate) fn wait(mut self) -> Result<ExitStatus, CommandError> {
-        self.process.wait().map_err(CommandError::from)
-    }
-
     pub fn get_pid(&self) -> u32 {
         self.process.id()
     }
 
+    pub fn shutdown(&mut self) -> Result<(), CommandError> {
+        let pid = self.get_pid() as i32;
+
+        use nix::{sys::signal, unistd::Pid};
+        signal::kill(Pid::from_raw(pid), signal::SIGTERM)
+            .map_err(|err| CommandError::NixError(err.to_string()))?;
+
+        if self.is_running_after_timeout(Duration::from_secs(10)) {
+            signal::kill(Pid::from_raw(pid), signal::SIGKILL)
+                .map_err(|err| CommandError::NixError(err.to_string()))?;
+        }
+        Ok(())
```

Comment:
After the timeout we send the kill signal again, but we actually don't know if the process is actually stopped. Is it OK to return `Ok(())`?